BuildScript/BuildTasks/MetadataValueDictionary.cs

```diff
@@ -67,5 +67,22 @@ public bool ContainsKey (string key)
     {
       return _rawDictionary.ContainsKey (key);
     }
+
+    public IEnumerable<KeyValuePair<string, string>> GetAllMatches (IEnumerable<string> possibleKeys, Func<Exception> createEmptySequenceException)
```

Comment:
I don't like the name GetAllMatches, but I don't have a better one right now.

Reply:
I've renamed it to `Intersect`, I believe it's symmetrical to `Single`